src/pages/Profile/ChangePassword/index.jsx

```diff
@@ -1,88 +1,80 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
-import React, { useState, useEffect } from 'react';
+import React, { useEffect } from 'react';
 import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers';
+import * as yup from 'yup';
+import classNames from 'classnames';
 
 import { changeUserPassword, authCleanUp } from 'state/actions/auth';
-import { useChangeHandler, useFormatMessage } from 'hooks';
+import { useFormatMessage } from 'hooks';
+import ErrorMessage from 'components/ErrorMessage';
+
+const schema = yup.object().shape({
+  current: yup.string().min(6).required(),
+  new: yup
+    .string()
+    .min(6)
+    .notOneOf([yup.ref('current')])
+    .required(),
+  confirmation: yup
+    .string()
+    .equals([yup.ref('new')])
+    .required(),
+});
 
 const ChangePasswordCard = () => {
-  const [passwords, setPasswords] = useState({
-    current: '',
-    new: '',
-    confirmation: ''
-  });
-
-  const onChangeHandler = useChangeHandler(setPasswords);
-
   const { loading, changedPassword } = useSelector(
-    state => ({
+    (state) => ({
       loading: state.auth.loading,
-      changedPassword: state.auth.changedPassword
+      changedPassword: state.auth.changedPassword,
     }),
     shallowEqual
   );
 
   const dispatch = useDispatch();
 
-  useEffect(() => {
-    return () => dispatch(authCleanUp());
-  }, [dispatch]);
+  const { register, handleSubmit, watch, setValue, errors } = useForm({
+    defaultValues: {
+      current: '',
+      new: '',
+      confirmation: '',
+    },
+    resolver: yupResolver(schema),
+  });
 
   useEffect(() => {
     if (changedPassword) {
-      setPasswords({
-        current: '',
-        new: '',
-        confirmation: ''
-      });
+      setValue('current', '');
+      setValue('new', '');
+      setValue('confirmation', '');
     }
-  }, [changedPassword, setPasswords]);
+    return () => dispatch(authCleanUp());
+  }, [dispatch, changedPassword]);
 
-  let inputs = {
-    new: {
-      modifier: null,
-      message: { modifier: null, content: null }
-    },
-    confirmation: {
-      modifier: null,
-      message: { modifier: null, content: null }
-    }
-  };
+  const newPassword = watch('new');
+  const currentPassword = watch('current');
+  const confirmationPassword = watch('confirmation');
 
-  const setInputs = (key, value) => {
-    inputs = { ...inputs, [`${key}`]: value };
-  };
+  const isNewPasswordSecure = newPassword && newPassword.length >= 6;
 
-  const isNewPasswordSecure = passwords.new && passwords.new.length >= 6;
+  const isCurrentPasswordInvalid =
```

Comment:
Can we do realtime validations with yup / react-hook-form and not wait for a submit in order to validate? if we can, we should avoid using this manual validations